src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1864,37 +1871,23 @@ void TorrentImpl::handleFileRenamedAlert(const lt::file_renamed_alert *p)
     // Remove empty leftover folders
     // For example renaming "a/b/c" to "d/b/c", then folders "a/b" and "a" will
     // be removed if they are empty
-    const QString oldFilePath = m_filePaths.at(fileIndex);
-    const QString newFilePath = Utils::Fs::toUniformPath(p->new_name());
+    const Path oldFilePath = m_filePaths.at(fileIndex);
+    const Path newFilePath {QString(p->new_name())};
 
     // Check if ".!qB" extension was just added or removed
-    if ((oldFilePath != newFilePath) && (oldFilePath != newFilePath.chopped(QB_EXT.size())))
+    // We should compare path in a case sensitive manner even on case insensitive
+    // platforms since it can be renamed by only changing case of some character(s)
+    if ((oldFilePath.data() != newFilePath.data())
+            && ((oldFilePath + QB_EXT) != newFilePath))
     {
         m_filePaths[fileIndex] = newFilePath;
 
-        QList<QStringView> oldPathParts = QStringView(oldFilePath).split('/', Qt::SkipEmptyParts);
-        oldPathParts.removeLast();  // drop file name part
-        QList<QStringView> newPathParts = QStringView(newFilePath).split('/', Qt::SkipEmptyParts);
-        newPathParts.removeLast();  // drop file name part
-
-#if defined(Q_OS_WIN)
-        const Qt::CaseSensitivity caseSensitivity = Qt::CaseInsensitive;
-#else
-        const Qt::CaseSensitivity caseSensitivity = Qt::CaseSensitive;
-#endif
-
-        int pathIdx = 0;
-        while ((pathIdx < oldPathParts.size()) && (pathIdx < newPathParts.size()))
-        {
-            if (oldPathParts[pathIdx].compare(newPathParts[pathIdx], caseSensitivity) != 0)
-                break;
-            ++pathIdx;
-        }
-
-        for (int i = (oldPathParts.size() - 1); i >= pathIdx; --i)
+        Path oldParentPath = oldFilePath.parentPath();
+        const Path commonBasePath = Path::commonPath(oldParentPath, newFilePath.parentPath());
+        while (oldParentPath != commonBasePath)
         {
-            QDir().rmdir(savePath() + Utils::String::join(oldPathParts, QString::fromLatin1("/")));
-            oldPathParts.removeLast();
+            Utils::Fs::rmdir(actualStorageLocation() / oldParentPath);
```

Comment:
The `oldParentPath` is already a full path, isn't? So is `actualStorageLocation()` still needed?

Reply:
> The `oldParentPath` is already a full path, isn't?

No, it isn't. Why did you think so?

Reply:
never mind.